pyvista/utilities/geometric_objects.py

```diff
@@ -726,21 +714,23 @@ def Pyramid(pointa, pointb, pointc, pointd, pointe):
     >>> pointc = [-1.0, -1.0, 1.0]
     >>> pointd = [1.0, -1.0, 1.0]
     >>> pointe = [0.0, 0.0, 0.0]
-    >>> pyramid = pyvista.Pyramid(pointa, pointb, pointc, pointd, pointe)
+    >>> pyramid = pyvista.Pyramid([pointa, pointb, pointc, pointd, pointe])
     >>> pyramid.plot() # doctest:+SKIP
     """
-    check_valid_vector(pointa, 'pointa')
-    check_valid_vector(pointb, 'pointb')
-    check_valid_vector(pointc, 'pointc')
-    check_valid_vector(pointd, 'pointd')
-    check_valid_vector(pointe, 'pointe')
+    assert(len(points) == 5)
```

Comment:
Better to have a `raise ValueError` here.  There's a good discussion for why we should raise errors instead of asserts here:
https://stackoverflow.com/a/20076295/3369879

In short, `assert` for testing, `raise` for program logic.